titus-server-runtime/src/main/java/com/netflix/titus/runtime/endpoint/v3/grpc/query/V3AbstractQueryCriteriaEvaluator.java

```diff
@@ -155,6 +157,19 @@ public boolean test(Pair<Job<?>, TASK_OR_SET> jobListPair) {
         return apply(imageTag, job -> job.getJobDescriptor().getContainer().getImage().getTag());
     }
 
+    private Optional<Predicate<Pair<Job<?>, TASK_OR_SET>>> applyPlatformSidecar(Optional<String> platformSidecarNameOpt) {
+        return platformSidecarNameOpt.map(platformSidecarName ->
+                jobTaskPair -> {
+                    List<PlatformSidecar> platformSidecars = jobTaskPair.getLeft().getJobDescriptor().getPlatformSidecars();
+                    List<String> psNames = new ArrayList<>();
+                    for (PlatformSidecar ps : platformSidecars) {
+                        psNames.add(ps.getName());
```

Comment:
You could make the comparison here, and return immediately if `ps.geName().equals(platformSidecarName)`

Reply:
Oh yea of course!